mentat/agent_handler.py

```diff
@@ -0,0 +1,155 @@
+import shlex
+from pathlib import Path
+from typing import List
+
+from openai.types.chat import (
+    ChatCompletionAssistantMessageParam,
+    ChatCompletionMessageParam,
+    ChatCompletionSystemMessageParam,
+)
+
+from mentat.code_feature import CodeMessageLevel
+from mentat.llm_api_handler import count_tokens, prompt_tokens
+from mentat.prompts.prompts import read_prompt
+from mentat.session_context import SESSION_CONTEXT
+from mentat.session_input import ask_yes_no, collect_user_input
+from mentat.transcripts import ModelMessage
+
+agent_file_selection_prompt_path = Path("agent_file_selection_prompt.txt")
+agent_command_prompt_path = Path("agent_command_selection_prompt.txt")
+
+
+class AgentHandler:
+    def __init__(self):
+        self._agent_enabled = False
+
+        self.agent_file_message = ""
+        self.agent_file_selection_prompt = read_prompt(agent_file_selection_prompt_path)
+        self.agent_command_prompt = read_prompt(agent_command_prompt_path)
+
+    # Make this property readonly because we have to set things when we enable agent mode
+    @property
+    def agent_enabled(self):
+        return self._agent_enabled
+
+    def disable_agent_mode(self):
+        self._agent_enabled = False
+
+    async def enable_agent_mode(self):
+        ctx = SESSION_CONTEXT.get()
+
+        self._agent_enabled = True
+        ctx.stream.send(
```

Comment:
A loading circle here would be nice haha

Reply:
Well we don't know how long it's going to take, so I'm not sure it's too applicable